hnn_core/network_builder.py

```diff
@@ -337,6 +337,8 @@ def __exit__(self, cell_type, value, traceback):
     # creates self._gid_list for THIS node
     def _gid_assign(self):
 
+        self.net._update_cells()  # updates net.n_cells
```

Comment:
cc @rythorpe

Reply:
Anticipating a public API where gids might change after instantiation?

Reply:
this was actually my fix for [this issue](https://github.com/jonescompneurolab/hnn-core/pull/357#discussion_r651264387). The problem was that `_update_cells` was called after gid assignment but gid assignment needs to know total number of cells which is updated only in `_update_cells`.